src/shell/scripts/omp.ps1

```diff
@@ -265,57 +269,56 @@ Example:
         $lastCommandSuccess = $?
         # store the last exit code for restore
         $realLASTEXITCODE = $global:LASTEXITCODE
-        $cleanPWD, $cleanPSWD = Get-PoshContext
         if ($script:TransientPrompt -eq $true) {
-            @(Start-Utf8Process $script:OMPExecutable @("print", "transient", "--error=$script:ErrorCode", "--pwd=$cleanPWD", "--pswd=$cleanPSWD", "--execution-time=$script:ExecutionTime", "--config=$env:POSH_THEME", "--shell-version=$script:PSVersion", "--shell=::SHELL::")) -join "`n"
+            $promptType = "transient"
             $script:TransientPrompt = $false
-            return
-        }
-        # see https://github.com/JanDeDobbeleer/oh-my-posh/issues/2483#issuecomment-1175761456
-        # and https://github.com/JanDeDobbeleer/oh-my-posh/issues/2502#issuecomment-1179968052
-        if (-not ((Get-PSCallStack).Location -join "").StartsWith("<")) {
-            @(Start-Utf8Process $script:OMPExecutable @("print", "debug", "--pwd=$cleanPWD", "--pswd=$cleanPSWD", "--config=$env:POSH_THEME", "--shell=::SHELL::")) -join "`n"
-            return
-        }
-        Initialize-ModuleSupport
-
-        $script:ExecutionTime = -1
-        $lastHistory = Get-History -ErrorAction Ignore -Count 1
-        if ($null -ne $lastHistory -and $script:LastHistoryId -ne $lastHistory.Id) {
-            $script:LastHistoryId = $lastHistory.Id
-            $script:ExecutionTime = ($lastHistory.EndExecutionTime - $lastHistory.StartExecutionTime).TotalMilliseconds
-            # error code should be changed only when a non-empty command is called
-            $script:ErrorCode = 0
-            if (!$lastCommandSuccess) {
-                $invocationInfo = try {
-                    # retrieve info of the most recent error
-                    $global:Error[0] | Where-Object { $_ -ne $null } | Select-Object -ExpandProperty InvocationInfo
-                } catch { $null }
-                # check if the last command caused the last error
-                if ($null -ne $invocationInfo -and $lastHistory.CommandLine -eq $invocationInfo.Line) {
-                    $script:ErrorCode = 1
-                } elseif ($realLASTEXITCODE -is [int] -and $realLASTEXITCODE -ne 0) {
-                    # native app exit code
-                    $script:ErrorCode = $realLASTEXITCODE
+        } else {
```

Comment:
I don't like to use if/else, makes everything rather difficult to read. So please refactor this to early returns. the whole block is now indented. We could also create an internal function which returns the prompt type.

Reply:
I have updated the refactoring. Also, I wrote the logic of getting the error code into a function and used early returns instead of "else" statements that you dislike. 😄